src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -948,7 +961,12 @@ void give_task_to_global_scheduler(LocalSchedulerState *state,
   DCHECK(state->config.global_scheduler_exists);
   Task *task = Task_alloc(spec, task_spec_size, TASK_STATUS_WAITING, NIL_ID);
   DCHECK(state->db != NULL);
-  task_table_add_task(state->db, task, NULL, NULL, NULL);
+  auto retryInfo = RetryInfo{
+      .num_retries = 0,  // This value is unused.
```

Comment:
if `num_retries` is unused, what's the behavior on failure? How many retries (and what controls that)?

Reply:
If `task_table_add_task` fails (i.e., the global scheduler has not started yet), then `give_task_to_global_scheduler_retry` will be called. Which will lead to another call to `task_table_add_task`. This will happen indefinitely until the call to `task_table_add_task` succeeds (or the local scheduler decides to keep the task locally).